### PR TITLE
Task 8 - Docker. AWS Elastic Beanstalk

### PR DESCRIPTION
**Pull request by tsialiukigor**
**### Task 8 - Docker. AWS Elastic Beanstalk**

**1.** Dockerfile is prepared, the image is building. Image size is minimized to be less than 500 MB.  **DONE**
image
**2.** Dockerfile is optimized. **DONE**
**3.** Folders are added to .dockerignore, with explanations. At least 2 big directories should be excluded from the build context. Elastic Beanstalk application is initialized. **DONE**
**4.** Environment is created and the app is deployed to the AWS cloud. **DONE**
cart-service-api link: http://cart-api-service-dev.eu-west-1.elasticbeanstalk.com/ping
**5.** FE application is updated with Cart API endpoint. **DONE**
fe commit: https://github.com/sijlver/nodejs-aws-fe/commit/4b51b71379865c836e8609ce9e8f320f1faf9e9b
fe link: http://d2w2h6xfwejks9.cloudfront.net/

**Estimated score: 5**

### CODE REVIEW

File: cart-api-service/.dockerignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+node_modules
+npm-debug.log
```
Comment (NikaOrl) [inline cart-api-service/.dockerignore]: Folders are added to .dockerignore, **with explanations**. **At least 2 big directories should be excluded from build context.** Elastic Beanstalk application is initialized.

Please, optimize your .dockerignore

File: cart-api-service/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
+FROM node:12-alpine
+
+# Create app directory inside the image
+WORKDIR /usr/src/app
+
+# Copy package.json and package-lock.json file from the current directory to the image
+COPY package*.json ./
+
+# Install node_modules
+RUN npm i && npm cache clean --force
```
Comment (NikaOrl) [inline cart-api-service/Dockerfile]: You're including dev dependencies by this way

**Review by NikaOrl — CHANGES_REQUESTED** [review]

**Comment by NikaOrl** [comment]
- [x]  Dockerfile is prepared, image is building. Image size is minimised to be less than 500 MB.
- [ ]  Dockerfile is optimized. Files that change more often and commands that depend on them should be included later, files and commands that change less should be at the top.
- [ ]  Folders are added to .dockerignore, with explanations. At least 2 big directories should be excluded from build context. Elastic Beanstalk application is initialized.
- [x] Environment is created and the app is deployed to the AWS cloud. You must provide a link to your GitHub repo with Cart API service or PR with created Dockerfile and related configurations.
- [x] FE application is updated with Cart API endpoint. You must provide a PR with updates in your FE repository and OPTIONALLY link to deployed front-end app which makes proper API calls to your Cart service.

I'm not sure that I'm right, but I'll score it like 3/5 for now

File: cart-api-service/Dockerfile (hunk 2 above)
Comment (tsialiukigor) [reply cart-api-service/Dockerfile]: Added the flag for production

File: cart-api-service/.dockerignore (hunk 1 above)
Comment (tsialiukigor) [reply cart-api-service/.dockerignore]: Added explanation and more files.

**Comment by webmakaka** [comment]
Привет!
По-любому придется настраивать работу сервиса по https.

Вот инструкция, если что.
https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-create-api-as-simple-proxy-for-http.html

Из-за хреново сформулированных требований, не вижу каких-либо причин что-то считать несделанным.

**Pull request merged**